modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        when(userManager.getGroup(id, requiredAttributes)).thenThrow(BadRequestException.class);
+        SCIMResponse scimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_BAD_REQUEST);
+    }
+
+    @Test(dataProvider = "dataForGetGroupNotExceptions")
+    public void testGetGroupNotImplementedException(String id, String attributes, String excludeAttributes)
+            throws CharonException, BadRequestException, NotFoundException, NotImplementedException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotImplementedException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+        when(userManager.getGroup(id, requiredAttributes)).thenThrow(NotImplementedException.class);
+        SCIMResponse scimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NOT_IMPLEMENTED);
+    }
+
+    @DataProvider(name = "dataForTestCreateGroupSuccess")
+    public Object[][] dataToTestCreateGroupSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        Group group = getNewGroup();
+
+        return new Object[][]{
+                {NEW_GROUP_SCIM_OBJECT_STRING, "id", null, group}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateGroupSuccess")
+    public void testCreateGroupSuccess(String scimObjectString, String attributes,
+                                       String excludeAttributes, Object objectGroup)
```

Comment:
casting is not required if we directly pass the param as `Group objectGroup` instead of passing it as an `Object`.